module/VuFind/src/VuFind/View/Helper/Root/ClassBasedTemplateRendererTrait.php

```diff
@@ -101,19 +96,34 @@ protected function resolveClassTemplate($template, $className,
      * @param string $template  Template path (with %s as class name placeholder)
      * @param string $className Name of class to apply to template.
      * @param array  $context   Context for rendering template
+     * @param bool   $throw     If true (default), an exception is thrown if the
+     * template is not found. Otherwise an empty string is returned.
      *
      * @return string
      * @throws RuntimeException
      */
-    protected function renderClassTemplate($template, $className, $context = [])
-    {
+    protected function renderClassTemplate($template, $className, $context = [],
+        $throw = true
+    ) {
         // Set up the needed context in the view:
         $view = $this->getView();
         $contextHelper = $view->plugin('context');
         $oldContext = $contextHelper($view)->apply($context);
 
         // Find and render the template:
-        $html = $view->render($this->getCachedClassTemplate($template, $className));
+        $classTemplate = $this->getCachedClassTemplate($template, $className);
```

Comment:
Maybe I'm not reading the code closely enough, but one concern: by moving the throw logic here, do we run the risk that if a template exists but contains no content, it will resolve as false and cause an exception to be thrown even when we do not want one? Would it be safer to keep the exception-throwing where it is in the current code, but wrap it in a try..catch here that returns empty string when $throw is false?

Reply:
getCachedClassTemplate just returns the template name, or empty string if it doesn't exist. The actual rendering happens in the next step, `$html = $view->render($classTemplate);`. But I see one problem now, it's not restoring the context properly. Fix coming up.

Reply:
Fix done. So to iterate, getCachedClassTemplate returns the template name, if found. Then we check if it's empty and an exception is needed. Otherwise render to $html if template exists, otherwise set it to an empty string.

Reply:
Thanks!